hazelcast/src/main/java/com/hazelcast/spi/impl/BasicOperationService.java

```diff
@@ -605,6 +608,24 @@ public void process(Operation op) {
             }
         }
 
+        private void traceSlowInvocation(Operation op) {
```

Comment:
I'm not seeing the full code, but this should not be a responsibility of the service. Imho the trouble shooting monitor should be able to look inside of components and then apply its own logic to do something with it. Thing like creating reports is a responsibility of the trouble shooting monitor, not of the component. I know.. I did the same with the performance monitor. But that also needs to be cleaned up.